lib/dal/src/func/authoring/save.rs

```diff
@@ -256,7 +256,13 @@ async fn update_leaf_associations(
     // for the provided components.
     id_set.extend(schema_variant_ids);
     for component_id in component_ids {
-        id_set.insert(Component::schema_variant_id(ctx, component_id).await?);
+        // TODO(nick): destroy nilId. Log a warning at the moment in case the frontend sends value
+        // for no-ops. I will come back and destroy nil id soon.
+        if component_id == ComponentId::NONE {
+            warn!("skipping component id set to nil id");
```

Comment:
In between now and when I am back on keyboard again, there's no reason to error if the frontend accidentally sends down the hellspawn that is `nilId` here. Just skip it or perish. If the frontend wanted a real `ComponentId`, it would have sent one down for leaf prototypes.